wandb/cli/cli.py

```diff
@@ -1519,6 +1515,194 @@ def scheduler(
+    pass
+
+
+@job.command("list")
+@click.option(
+    "--project",
+    "-p",
+    envvar=env.PROJECT,
+    help="The project you want to list jobs from.",
+)
+@click.option(
+    "--entity",
+    "-e",
+    default="models",
+    envvar=env.ENTITY,
+    help="The entity the jobs belong to",
+)
+def _list(project, entity):
+    wandb.termlog(f"Listing jobs in {entity}/{project}")
+    public_api = PublicApi()
+    try:
+        jobs = public_api.list_jobs(entity=entity, project=project)
+    except wandb.errors.CommError as e:
+        wandb.termerror(f"{e}")
+        return
+
+    if len(jobs) == 0:
+        wandb.termlog("No jobs found")
+        return
+
+    for job in jobs:
+        aliases = []
+        if len(job["edges"]) == 0:
+            # deleted?
+            continue
+
+        name = job["edges"][0]["node"]["artifactSequence"]["name"]
+        for version in job["edges"]:
+            aliases += [x["alias"] for x in version["node"]["aliases"]]
+
+        # only list the most recent 10 job versions
+        aliases_str = ",".join(aliases[::-1])
+        wandb.termlog(f"{name} -- versions ({len(aliases)}): {aliases_str}")
+
+
+@job.command()
+@click.argument("job")
+def describe(job):
+    public_api = PublicApi()
+    try:
+        job = public_api.job(name=job)
+    except wandb.errors.CommError as e:
+        wandb.termerror(f"{e}")
+        return
+
+    for key in job._job_info:
+        if key.startswith("_"):
+            continue
+        wandb.termlog(f"{key}: {job._job_info[key]}")
+
+
+@job.command(
+    no_args_is_help=True,
+)
+@click.option(
+    "--project",
+    "-p",
+    envvar=env.PROJECT,
+    help="The project you want to list jobs from.",
+)
+@click.option(
+    "--entity",
+    "-e",
+    default="models",
+    envvar=env.ENTITY,
+    help="The entity the jobs belong to",
+)
+@click.option(
+    "--name",
+    "-n",
+    help="Name for the job",
+)
+@click.option(
+    "--description",
+    "-d",
+    help="Description for the job",
+)
+@click.option(
+    "--alias",
+    "-a",
+    "aliases",
+    help="Alias for the job",
+    multiple=True,
+    default=tuple(),
+)
+@click.option(
+    "--entry-point",
+    "-E",
+    "entrypoint",
+    help="Codepath to the main script, required for repo jobs",
+)
+@click.option(
+    "--git-hash",
+    "-g",
+    "git_hash",
+    type=str,
+    help="Hash to a specific git commit.",
+)
+@click.option(
+    "--runtime",
+    "-r",
+    type=str,
+    help="Python runtime to execute the job",
+)
+@click.argument(
+    "job_type",
+    type=click.Choice(("git", "code", "image")),
+)
+@click.argument("path")
+def create(
+    path,
+    project,
+    entity,
+    name,
+    job_type,
+    description,
+    aliases,
+    entrypoint,
+    git_hash,
+    runtime,
+):
+    """Create a job from a source, without a wandb run.
+
+    Jobs can be of three types, git, code, or image.
+
+    git: A git source, with an entrypoint pointing to the main python executable.
```

Comment:
As a non-blocking low priority follow on, I'd like us to allow users to give a raw repo, then be able to optionally specify a git hash, and be forced to give the path to the entrypoint

Reply:
I think this happens automatically, i'll update the comment. We fail if we don't find an entrypoint, so if theres none in path then essentially we require it to be passed in.